### ISSUE TITLE
"New variable tweaks" addon for creating multiple variables at once plus other features

### ISSUE THREAD

**Issue by Hexagonl** [ScratchAddons/ScratchAddons#6742]
### Why this would be helpful

Oftentimes, bugs in projects occur due to people creating a new variable incorrectly. These QOL tweaks can mitigate them and make the variable creation process a lot faster. Also, these features would be useful when ported to other addons.

### How this addon works

![New Variable Modal Mockup C](https://github.com/ScratchAddons/ScratchAddons/assets/61996651/9e2f0e7f-9f3f-4809-95e3-7865de5f73e6)
### Proposed features
- Create multiple variables at once
- Set the initial value of a variable
- Predictive new variable name
  - Will detect when there is a number or letter X, Y, Z at the end of the variable, then suggest creating a new variable of the same name but incremented (for example: suggest creating "positionY" if "positionX" exists)

### Design
A table would be the most intuitive way to display this information to the user. See the mockup above to understand what I mean.

### Possible alternatives

These features (specifically, the table in the mockup) could also be ported to the `variable-manager` addon, which is due for an upgrade. Adding a row to create new variables (like the one in the mockup) would also be useful, which would solve #4911.

For continuity, if this addon were to be created, then the `swap-local-global` (change variable scope) addon might need to be updated to have this interface. The benefit of doing this is it'll make it easier to add a "multi-select variables" addon in the future.

Perhaps the `hide-new-variables` addon could benefit from having an accessible toggle on this new, new variable modal? Not sure if this is necessary, it's just an extra thought. If the functionality of these two addons eventually became intertwined, then perhaps that hide new variables addon could be merged into this one.

### Mockup info
- *Icons from feathericons.com; cloud is the cloud emoji since apparently that's what scratch uses; color/theme is default scratch*
- I designed the mockup in Figma and published it, so you can view it here in case you want the exact colors or to remix it:
https://www.figma.com/community/file/1291257436821669218/scratch-new-variable-modal-mockup

These features were suggested in the Scratch Addons community discord, here are links to the threads:
[Create multiple variables at once](https://discord.com/channels/806602307750985799/1143301502920835163)
[Set the value of a new variable](https://discord.com/channels/806602307750985799/1157551618598899772)
[*Predictive new variable name*](https://discord.com/channels/806602307750985799/1143301502920835163/1143606384865329222)

**Comment by Samq64** [comment]
Thank for the detailed feature request! I think this would fit best in the variable manager but it would also work well as a separate addon based on your mockup.

**Comment by DNin01** [comment]
Wow! Those are quite a few ideas!

I can see how creating multiple new variables at the same time might be helpful, maybe? Oh wait, variable name suggestions? What?!?! OMG. Setting the starting value from there could be useful for some situations too.

Editing multiple at once sounds really neat, too!

**Comment by DNin01** [comment]
Navigation via keyboard within the modal is also worth considering.

**Comment by Hexagonl** [comment]
> Imagine if you just clicked "Cancel" after all that… (we should consider confirming before closing the modal)

Another solution is to make the modal’s data persist even after you close it. Reopening the modal would put back the same data. It would only clear if you press OK or pressed a “Clear” button instead

**Comment by medlans** [comment]
You actually used to be able to kind of do this in Scratch 2.0 to create multiple vars of the same scope:
1. Create a variable of a specific scope (make sure to press enter to do so instead of clicking the ok button)
2. It's hard to explain, but it acts as if the previous name is still there, so basically you can backspace until it's "gone" (aka just backspace for the length of the previous variable's name or longer), then type the new name, or just replace some of the letters if it's similarly named.
3. Press enter again. It should create a variable of the previous scope.
4. Repeat 2-3 for every variable you want to create.

It worked with lists and custom blocks too, so I'm thinking that this should also work with those (I honestly think they should've added something like this in 3.0 just so others could have saved time).

**Comment by medlans** [comment]
> It worked with lists and custom blocks too, so I'm thinking that this should also work with those (I honestly think they should've added something like this in 3.0 just so others could have saved time).

Yeah, maybe this could also work for editing multiple variables/lists/and maybe More Blocks (so changing their name or scope or whatever) at the same time? I was thinking of this.
